### ISSUE TITLE
torchmetrics Accuracy

### ISSUE THREAD

**Issue by Antoine101** [Lightning-AI/torchmetrics#2555]
### 📚 Documentation

Hi Lightning team,

I was looking at metrics and found something a tad confusing.

In the following screenshot, the accuracy metric is imported like so:
`torchmetrics.classification.Accuracy`
![image](https://github.com/Lightning-AI/pytorch-lightning/assets/48209504/b574fcb4-bc84-4703-a3cb-9e9a71d16422)

But looking at this other page of the Torchmetrics doc, it shows: `torchmetrics.Accuracy` (without `classification` in between)
![image](https://github.com/Lightning-AI/pytorch-lightning/assets/48209504/1bd392dc-431b-4afb-a9f5-6e06620b40ed)

In my script, the following work:
`import torchmetrics.Accuracy`
`import torchmetrics.classification.accuracy`

But not the following:
`import torchmetrics.classification.Accuracy`

Which one is the right one?

If it is `torchmetrics.Accuracy`, why do the other classification metrics seem to require the `classification` in between?
![image](https://github.com/Lightning-AI/pytorch-lightning/assets/48209504/ede854fa-bcb7-469c-be39-6b9e85b4c693)

Cheers

Antoine

cc @borda

**Comment by Borda** [comment]
> torchmetrics.Accuracy ✅
> torchmetrics.classification.accuracy ✅
> torchmetrics.classification.Accuracy ❌

I can reproduce it, but on the other hand, this works:
`from torchmetrics.classification import Accuracy`

**Comment by Antoine101** [comment]
Sorry, all three work actually on my side as well!

But how do they all differ?

**Comment by Borda** [comment]
> But how do they all differ?

Tbh, not sure...

**Comment by SkafteNicki** [comment]
Let me try to clarify

```import torchmetrics.Accuracy```
should not work. Reason being that there is no module/file in `src/torchmetrics/*` called `Accuracy`. Instead If you do
```from torchmetrics import Accuracy```
it will not look for a file called `Accuracy.py` but instead it looks in the `src/torchmetrics/__init__.py` file for a reference to a metric called `Accuracy`, which it gets from these lines of code: https://github.com/Lightning-AI/torchmetrics/blob/master/src/torchmetrics/__init__.py.

```import torchmetrics.classification.accuracy```
this should work, but you are not getting the `accuracy` metric you are instead importing the `src/torchmetrics/classification/accuracy.py` file which contains the definition to `Accuracy` metric. There is a difference between importing the file and importing the function/module you want.

```import torchmetrics.classification.Accuracy```
again should not work because there is no file called `src/torchmetrics/classification/Accuracy.py`. Instead if we do
```from torchmetrics.classification import Accuracy```
you will get the same `Accuracy` metric as in the first case.

The very short answer is: the recommendation is to ALWAYS import metrics from there respective domain e.g.
```python
from torchmetrics.classification import Accuracy, Precision, Recall
```
or use full path references
```python
import torchmetrics
metric = torchmetrics.classification.Accuracy(...)
```

however, some metrics can still be directly access from the root of torchmetrics e.g. `torchmetrics.Accuracy` and that is mostly due to historical reasons and to keep things backwards compatible. But that does not mean that there are multiple Accuracy metrics, just that there are different ways to import the same metric.

Closing issue.
